Change IsZ_pi from a property to a category

fingolfin opened the pull request:
This seems correct, as there were no real methods to compute it. Also, it is very similar to IsIntegers, which is also a category.

This makes a so-called "hidden" implication created by DeclareProperty explicit, thereby resolving a bunch of warnings that show up if one starts the upcoming GAP 4.11 with the `-N` command line option, and then loads this package.

For some information on the background of this, see also <https://github.com/gap-system/gap/issues/1649> and <https://github.com/gap-system/gap/issues/2336>

Stefan-Kohl commented:
@fingolfin Travis says "All checks have failed" ...

fingolfin commented:
Yes, they failed, due to a minor omission, which I fixed (I rely on Travis to run tests, which it did, quite reliably). It still fails in the GAP 4.9.3 test, as for some reason, it fails to load the "grape" package there -- but only in the part of the tests where it tests RCWA. This is really baffling to me.

olexandr-konovalov commented:
At the moment it timeouts in `rcwa/tst/cscrct.tst` under GAP 4.9.3. Of course, you can always decide to require GAP >= 4.10...

fingolfin commented:
Figured it out: The "All checks have failed" happened because the code I added to `.travis.yml` to run the RCWA test suite was buggy, resulting in RCWA not being loaded in GAP 4.9, and an subsequent infinite loop. This is now fixed. In the meantime, I also decided to completely redo this PR, so it's now no longer about a trivial implication, but rather changes a property to a category.

fingolfin committed:
Change IsZ_pi from a property to a category

This seems correct, as there were no real methods to compute it. Also,
it is very similar to IsIntegers, which is also a category.

This makes a so-called "hidden" implication created by DeclareProperty
explicit, thereby resolving a bunch of warnings that show up if one
starts the upcoming GAP 4.11 with the `-N` command line option, and then
loads this package.

For some information on the background of this, see also
<https://github.com/gap-system/gap/issues/1649> and
<https://github.com/gap-system/gap/issues/2336>
